### PR TITLE
Add priority 2 Tracks events

### LINKED ISSUE #2604: Tracks: Add priority 2 events
Add the following Tracks events to the UI

Event | Description | Props | Priority | Status
-- | -- | -- | -- | --
wcadmin_datepicker_update | When a user modifies the date range | type: preset vs custom, compare_to, preset ( if applicable ), length of date range if custom, page ( report name or dashboard ) | 2 | #2525
wcadmin_navbar_breadcrumb_click | When a link in the header breadcrumb is clicked | href | 2
wcadmin_analytics_table_header_toggle | When a table column is toggled | report name, column name, status on\|off | 2
wcadmin_analytics_table_sort | When the sort of a table is changed | report name, column name, direction ASC \| DESC | 2
wcadmin_analytics_filters_all_any | When the all/any filter drop-down is changed | report name, value: all or any | 2
wcadmin_analytics_table_filter | When a search is performed via the table filter area ( products, categories, customers, coupons, taxes ) | report name | 2

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #2604.

Introduces the following new Tracks events (`wcadmin_datepicker_update ` was already implemented):

Event | Description | Props | Priority | Status
-- | -- | -- | -- | --
`wcadmin_navbar_breadcrumb_click` | When a link in the header breadcrumb is clicked | href | 2
`wcadmin_analytics_table_header_toggle` | When a table column is toggled | report name, column name, status on\|off | 2
`wcadmin_analytics_table_sort` | When the sort of a table is changed | report name, column name, direction ASC \| DESC | 2
`wcadmin_analytics_filters_all_any` | When the all/any filter drop-down is changed | report name, value: all or any | 2
`wcadmin_analytics_table_filter` | When a search is performed via the table filter area ( products, categories, customers, coupons, taxes ) | report name | 2

In order to accomplish some of these events, props were added to the underlying components to accept "on action" functions.

### Detailed test instructions:

- Turn on tracks debug: `localStorage.setItem( 'debug', 'wc-admin:*' );`
- Perform the actions described in the table above
- Verify the tracks events are generated with the appropriate data

### CODE REVIEW

File: packages/components/src/table/index.js [hunk 1]
```diff
@@ -522,6 +537,8 @@ TableCard.defaultProps = {
 	isLoading: false,
 	onQueryChange: noop,
 	onColumnsChange: noop,
+	onSearch: noop,
```
Comment (timmyc) [inline packages/components/src/table/index.js]: nice.

File: packages/components/src/table/index.js [hunk 2]
```diff
@@ -179,8 +179,13 @@ class TableCard extends Component {
 		}
 	}
 
-	onSearch( values ) {
-		const { compareParam, searchBy, baseSearchQuery } = this.props;
+	onSearchChange( values ) {
```
Comment (timmyc) [inline packages/components/src/table/index.js]: Thanks for taking care here and making sure to not leak tracks logic into the npm packages 👍

File: packages/components/src/table/index.js [hunk 3]
```diff
@@ -453,6 +460,14 @@ TableCard.propTypes = {
 	 * A function which returns a callback function which is called upon the user changing the visiblity of columns.
 	 */
 	onColumnsChange: PropTypes.func,
+	/**
+	 * A function which is called upon the user searching in the table header.
+	 */
+	onSearch: PropTypes.func,
```
Comment (timmyc) [inline packages/components/src/table/index.js]: With the addition of these two new props, I think we should note the changes in the "unreleased" section of the components package.

Comment (jeffstieler) [reply packages/components/src/table/index.js]: Done in 7030953.

**Comment by jeffstieler** [comment]
> `wcadmin_navbar_breadcrumb_click` is working great on wc-admin pages, but I'm not seeing the event logged, or a t.gif fired when clicking on breadcrumb links in embedded pages ( i.e. Settings )

I'm seeing the event get logged in console before the navigation occurs. You can use the gear icon on the right side of Chrome's console to enable "preserve log" so you can see console entries after a navigation event.

**Comment by timmyc** [comment]
> You can use the gear icon on the right side of Chrome's console to enable "preserve log" so you can see console entries after a navigation event.

Hmm I have that ticked in firefox, but will test again.

**Comment by jeffstieler** [comment]
In Firefox:

Screen Shot 2019-07-12 at 9 56 12 AM

**Pull request merged**